solar_system.rb

```diff
@@ -0,0 +1,352 @@
+  end
+
+  def orbit_period
+    return "The orbit period of #{@name} is #{@orbit_period}."
+  end
+
+  def length_of_day
+    return "The solar days of #{@name} are #{@length_of_day} long."
+  end
+
+  def distance_from_sun
+    return "#{@name} is #{@distance_from_sun} from the Sun."
+  end
+
+  def all_facts
+    return "#{self.planet_name} \n#{self.number_of_moons} \n#{self.surface_temp_c} \n#{self.diameter} \n#{self.orbit_period} \n#{self.length_of_day} \n#{self.distance_from_sun}"
+  end
+
+end
+
+class SolarSystem
+
+  def initialize (planets)
+    return @planets = planets
+  end
+
+  attr_accessor :planets
+
+  def planet
+    @planets << Planet.new
+  end
+
+  def system
+    return "#{@planets}"
+  end
+
+  def planet_list
+    puts "Our Solary System is made up of the following planets: "
+    planet_name = ""
+    planet_num = 0
+    @planets.each do |planet|
+      planet_num += 1
+      planet_name << "#{planet_num}. #{planet.name} \n"
+    end
+    return planet_name
+  end
+
+  def planet_facts
+    puts "Here are some facts about planets:"
+    @planets.each do |planet|
+      puts "#{@planets}"
+    end
+  end
+
+end
+
+planets = []
+
+jupiter = {:name => "Jupiter", :moons => 67, :surface_temp_c => -148, :diameter => "142984 km", :orbit_period => "4333 days", :length_of_day => "10 hours " }
+mars = {:name => "Mars", :moons => 2, :surface_temp_c => "-153 to 20" , :diameter => "6792 km", :orbit_period => "687 days", :length_of_day => " 25 hours" }
+venus= {:name => "Venus", :moons => 0, :surface_temp_c => 426, :diameter => "12104 km", :orbit_period => "225 days", :length_of_day => "117 earth days"}
+
+planets << Planet.new(jupiter)
+planets << Planet.new(mars)
+planets << Planet.new(venus)
+
+outer_space = SolarSystem.new(planets)
+
+puts "Welcome the Solar System app!"
+puts outer_space.planet_list
+
+interactions = ["EXPLORE","NEW","EXIT"]
+
+run_once = false
+interact = ""
+
+until interact == "EXIT"
+
+  # IF this is the first time the user is prompted to enter an action
+  # OR they chose a valid options THEN display the regular prompts
+  if (run_once == false || interactions.include?(interact))
+
+    if (run_once == false)
+      puts "To learn more about a specific planet type: Explore"
+      puts "Discovered a new planet? Type: New"
+      puts "Had enough? To quit type: Exit"
+    else
+      puts "What would you like to do (new, explore, exit)?"
+    end
+
+  # ELSE the loop has already run and the user supplied invalid input,
+  # so we need to prompt them to enter something valid.
+  else
+
+    puts "Please enter a valid option."
+
+  end
+
+  interact = gets.chomp.upcase
+
+  run_once = true
+
+  case interact
+    when "EXPLORE"
+      puts outer_space.planet_list
+      puts "Type the number that corresponds to the planet you would like to learn more about:"
+
+      # TO DO: Add a check to ensure they've supplied a valid number
+
+      puts planets[gets.chomp.to_i - 1].all_facts
```

Comment:
the value of `planets` is the same as `outer_space.planets`. It would be better if you used that array instead, so it was `outer_space.planets[gets.chomp.to_i - 1].all_facts`